src/fparser/two/Fortran2003.py

```diff
@@ -7787,23 +7787,35 @@ class Block_Label_Do_Construct(BlockBase):  # pylint: disable=invalid-name
     subclass_names = []
     use_names = ["Label_Do_Stmt", "Execution_Part_Construct", "End_Do"]
 
-    @staticmethod
-    def match(reader):
+    @classmethod
+    def match(cls, reader):
         """
         :param reader: instance of `FortranReaderBase` class
         :type reader: :py:class:`FortranReaderBase`
         :return: code block matching the labeled "DO" construct
         :rtype: string
         """
         return BlockBase.match(
-            Label_Do_Stmt,
+            cls.label_do_stmt_cls(),
             [Execution_Part_Construct],
             End_Do,
             reader,
             match_labels=True,
             enable_do_label_construct_hook=True,
         )
 
+    @staticmethod
+    def label_do_stmt_cls():
+        """
+        :returns: Fortran2003 Label_Do_Stmt class.
+        :rtype: :py:class:`fparser.two.Fortran2003.Label_Do_Stmt`
+
+        """
+        # Import here to avoid circular imports
```

Comment:
Please disable the pylint warning.

Reply:
I'm not sure why the import was added as the class is in the same file. Removed.